lib/b2b/passwords.ts

```diff
@@ -0,0 +1,221 @@
+}
+
+export interface EmailResetRequest {
+  password_reset_token: string;
+  password: string;
+  session_token?: string;
+  session_jwt?: string;
+  session_duration_minutes?: number;
+  session_custom_claims?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  code_verifier?: string;
+}
+
+export interface EmailResetResponse extends BaseResponse {
+  member_id: string;
+  member_email_id: string;
+  organization_id: string;
+  member: Member;
+  session_token?: string;
+  session_jwt?: string;
+  member_session?: MemberSession;
+  organization: Organization;
+}
+
+export interface ExistingPasswordResetRequest {
+  email_address: string;
+  existing_password: string;
+  new_password: string;
+  session_token?: string;
+  session_jwt?: string;
+  session_duration_minutes?: number;
+  session_custom_claims?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  organization_id: string;
+}
+
+export interface ExistingPasswordResetResponse extends BaseResponse {
+  member_id: string;
+  organization: Organization;
+  member: Member;
+  session_token?: string;
+  session_jwt?: string;
+  member_session?: MemberSession;
+}
+
+export interface SessionResetRequest {
+  password: string;
+  organization_id: string;
+  session_token?: string;
+  session_jwt?: string;
+}
+
+export interface SessionResetResponse extends BaseResponse {
+  member_id: string;
+  organization: Organization;
+  member: Member;
+  member_session?: MemberSession;
+}
+
+export interface StrengthCheckRequest {
+  email_address?: string;
+  password: string;
+}
+
+export interface StrengthCheckResponse extends BaseResponse {
+  valid_password: boolean;
+  score: number;
+  breached_password: boolean;
+  strength_policy: string;
+  breach_detection_on_create: boolean;
+  zxcvbn_feedback: {
+    suggestions: string[];
+    warning: string;
+  };
+  luds_feedback: {
+    has_lower_case: boolean;
+    has_upper_case: boolean;
+    has_digit: boolean;
+    has_symbol: boolean;
+    missing_complexity: number;
+    missing_characters: number;
+  };
+}
+
+interface MigrateRequestBase {
+  organization_id: string;
+  email_address: string;
+  hash: string;
+  name?: string;
+  trusted_metadata?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  untrusted_metadata?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+export type MigrateRequest =
+  | (shared.MD5MigrateRequest & MigrateRequestBase)
+  | (shared.BcryptMigrateRequest & MigrateRequestBase)
+  | (shared.Argon2IMigrateRequest & MigrateRequestBase)
+  | (shared.Argon2IDMigrateRequest & MigrateRequestBase)
+  | (shared.SHA1MigrateRequest & MigrateRequestBase)
+  | (shared.ScryptMigrateRequest & MigrateRequestBase)
+  | (shared.PHPassMigrateRequest & MigrateRequestBase);
+
+export interface MigrateResponse extends BaseResponse {
+  member_id: string;
+  organization_id: string;
+  member_created: boolean;
```

Comment:
Missing Member and Org fields - perhaps those should be pulled into their own `ResponseWithMember` interface that we can extend everywhere?

Reply:
I like it. It looks like we can include `member_id` there too

Reply:
SessionsAuthenticate differs but all of discovery and magic_links also include member_id